app/controllers/application_controller.rb

```diff
@@ -20,6 +20,25 @@ class ApplicationController < ActionController::Base
 
   layout 'application'
 
+  # A common method for controllers to find a Dor object.  It accepts any valid
+  # PID (druid:aa111aa1111) or DRUID (aa111aa1111).  DruidTools::Druid parses
+  # input and returns a uniform PID, which is input to Dor.find to obtain a Dor
+  # object.
+  # @see https://github.com/sul-dlss/druid-tools
+  # @param druid [String]
+  # @return object [ActiveFedora::Base] any Dor object
+  # @raises ArgumentError
+  def find_druid(druid)
```

Comment:
This method is vexing and requires more discussion.  It has evoled this way to provide (a) consistent behavior and (b) satisfy existing specs.  My gut feeling is that dor-services could use DruidTools in the Dor.find method and that would help to resolve this, but it wouldn't resolve the consistent handling of the exceptions.